api/controllers/FlairController.js

```diff
@@ -118,6 +133,16 @@ module.exports = {
       var pFlair = _.get(req, "user.flair.ptrades.flair_css_class") || "default";
       var svFlair = _.get(req, "user.flair.svex.flair_css_class") || "";
       svFlair = svFlair.replace(/2/, "");
+      
+      console.log(flairs);
+        
+      // For safety, remove all colons from flair and reset emoji.
+      let ptrades_current_text = flairs.ptrades.replace(/:[a-zA-Z0-9_-]*:/g,'');
```

Comment:
The server currently isn't validating whether the user's IGNs have a colon (as far as I can tell the validation is only clientside), so a user might be able to cause some weird behavior here if they put a colon in the middle of their flair text.

It might be better to have the client remove all emojis before sending the flair text to the server, and then have the server just reject a user-provided flair text if it contains a colon.

Reply:
Okay, I'm not completely sure but I think I did it correctly this time!
- Changed flairCheck in Flairs.js to include a check for emojis and throw an error
- Changed $scope.setFlairText in userCtrl.js to remove emojis from flair text before sending to the server